packages/devextreme/js/__internal/scheduler/m_compact_appointments_helper.ts

```diff
@@ -181,4 +184,29 @@ export class CompactAppointmentsHelper {
       .append($('<span>').text(text))
       .addClass(APPOINTMENT_COLLECTOR_CONTENT_CLASS);
   }
+
+  _localizeDate(date) {
+    return `${dateLocalization.format(date, 'monthAndDay')}, ${dateLocalization.format(date, 'year')}`;
+  }
+
+  _getStartDate(appointment) {
+    const date = appointment.startDate;
+    return date ? new Date(date) : null;
+  }
+
+  _getEndDate(appointment) {
+    const date = appointment.endDate;
+    return date ? new Date(date) : null;
+  }
+
+  _getDateText(appointment) {
+    const startDateText = this._localizeDate(this._getStartDate(appointment));
+    const endDateText = this._localizeDate(this._getEndDate(appointment));
+
+    const dateText = startDateText === endDateText
```

Comment:
@tongsonbarbs what would happen if both of the dates would be `null`?

Reply:
Thanks for noticing this, I tested it and it is returning undefined as string.. I have committed the changes now to fix this..